In [3]:
import pandas as pd
import helper_data_load as helper

# Sevinj calculation of short and long term classification os_days

In [112]:
path='../../Data/'

ch_cl=pd.read_csv(path+'clinical_CHUM.csv')
chum=pd.read_csv(path+'PyRads_CHUM.csv')

new_d=pd.merge(ch_cl,chum, on='oncotech_id')


In [119]:
a=new_d.os_days.quantile(0.333)
b=new_d.os_days.quantile(0.666)
c=new_d.os_days.quantile(0.666)
print("Q1 quantile: ", new_d.os_days.quantile(0.33))
print("Q2 quantile: ", new_d.os_days.quantile(0.66))
print("Q3 quantile: ", new_d.os_days.quantile(0.66))
print(a,b,c)


Q1 quantile:  280.78000000000003
Q2 quantile:  594.1200000000001
Q3 quantile:  594.1200000000001
282.778 602.964 602.964


In [117]:
df1=new_d[new_d.os_days<=a].reset_index()
df2=new_d[(new_d.os_days>a) & (new_d.os_days<=b)].reset_index()
df3=new_d[new_d.os_days>=b].reset_index()

print(len(df1), len(df3))

74 75


In [89]:
df1.os_days.sort_values()

2       9
60      9
26     14
47     16
36     17
     ... 
37    272
65    273
18    275
73    279
40    280
Name: os_days, Length: 74, dtype: int64

In [90]:
df2.os_days.sort_values()

66    283
54    285
30    286
60    293
69    294
     ... 
32    566
7     568
11    580
35    591
24    597
Name: os_days, Length: 74, dtype: int64

In [91]:
df3.os_days.sort_values()

13     604
63     621
67     624
21     641
24     662
      ... 
46    1760
16    1818
3     1888
58    2047
31    2047
Name: os_days, Length: 75, dtype: int64

In [92]:
#rad_path_chum='/Users/sevinjyolchuyeva/Desktop/postdoc/data_os_days/data/'
column_name='class_os_days'
#with open(rad_path_chum+"feature_radimics.json", "r") as fp:
#    feature_name = json.load(fp)

radomics_first = df1
radomics_second = df3
radomics_first[column_name] = 0
radomics_second[column_name] = 1
radomics_chum=pd.concat([radomics_first, radomics_second], ignore_index=True)
radomics_chum=radomics_chum.sample(frac=1).reset_index(drop=True)


## Sevinj caluclation but for genomics

In [30]:
def Sevinj_method_class(clinical, features_df):
    path='../../Data/'

    ch_cl=pd.read_csv(path + clinical + '.csv')
    chum_gen=pd.read_csv(path + features_df + '.csv')
    chum_gen = helper.prep_genomics(chum_gen)
    genetic_features = chum_gen.columns[1:]
    new_d=pd.merge(ch_cl,chum_gen, on='oncotech_id')
    print(len(new_d))
    a=new_d.os_days.quantile(0.333)
    b=new_d.os_days.quantile(0.666)
    c=new_d.os_days.quantile(0.666)
    print("Q1 quantile: ", new_d.os_days.quantile(0.33))
    print("Q2 quantile: ", new_d.os_days.quantile(0.66))
    print("Q3 quantile: ", new_d.os_days.quantile(0.66))
    print(a,b,c)
    df1=new_d[new_d.os_days<=a].reset_index()
    df2=new_d[(new_d.os_days>a) & (new_d.os_days<=b)].reset_index()
    df3=new_d[new_d.os_days>=b].reset_index()

    print(len(df1), len(df3))
    column_name='class_os_days'

    gen_first = df1
    gen_second = df3
    gen_first[column_name] = 0
    gen_second[column_name] = 1
    gen_chum=pd.concat([gen_first, gen_second], ignore_index=True)
    gen_chum=gen_chum.sample(frac=1).reset_index(drop=True)
    return gen_chum, genetic_features

In [33]:
gen_chum, genetic_features = Sevinj_method_class("clinical_CHUM", "Discovery-GenomicsTop5000-March22023")
gen_chum[list(genetic_features)+["class_os_days", "oncotech_id"]].to_csv("../../Data/CHUM_genetics_class.csv")


72
Q1 quantile:  296.34
Q2 quantile:  650.66
Q3 quantile:  650.66
304.434 661.864 661.864
24 24


In [34]:
gen_iucpq, genetic_features = Sevinj_method_class("clinical_IUCPQ", "ValidationIUCPQ-March2 2023")
gen_iucpq[list(genetic_features)+["class_os_days", "oncotech_id"]].to_csv("../../Data/IUCPQ_genetics_class.csv")


59
Q1 quantile:  405.0
Q2 quantile:  845.56
Q3 quantile:  845.56
405.0 846.256 846.256
21 20


# Elena's calculation of classification data

In [93]:
import helper_data_load

In [94]:
local = True
predictor = "os_days"
classification = True

data_norm_chum, data_norm_iucpq = helper_data_load.get_norm_data(local, predictor, classification)

223 patients with clinical and radiation data for CHUM.
162 patients with clinical and radiation data for IUCPQ.
223 patients after drop NaN - CHUM.
162 patients after drop NaN - IUCPQ.
149 patients with clinical and radiation data for CHUM and classification.
108 patients with clinical and radiation data for IUCPQ and classification.


/Users/iucpq/Documents/IUCPQ/IUCPQ/Predictions/helper_data_load.py:42: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  data_norm.columns = data_norm.columns.str.replace('.','_')
/Users/iucpq/Documents/IUCPQ/IUCPQ/Predictions/helper_data_load.py:42: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  data_norm.columns = data_norm.columns.str.replace('.','_')
/Users/iucpq/Documents/IUCPQ/IUCPQ/Predictions/helper_data_load.py:48: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["class_"+ predictor][df["

## Without function and normalization

In [95]:
import numpy as np

In [96]:
def add_os_day_class(df, predictor):
    df["class_"+ predictor] = pd.qcut(df[predictor], 3, labels=False, retbins=False, precision=3, duplicates='raise')
    df = df[df["class_"+ predictor] != 1]
    df["class_"+ predictor][df["class_"+ predictor] == 2] = 1
    df["class_"+ predictor][df["class_"+ predictor] == "nan"] = np.nan
    df = df[df["class_"+ predictor].notna()]
    
    df = df.reset_index(drop=True)
    #df = df.drop(columns=[predictor, "Unnamed: 0"])
    df["class_"+ predictor] = df["class_"+ predictor].astype('int')
    return df

In [97]:
pyrads_chum = pd.read_csv("../../Data/PyRads_CHUM.csv")
clinical_chum = pd.read_csv("../../Data/clinical_CHUM.csv")

chum_merge = pyrads_chum.merge(clinical_chum[["oncotech_id",predictor]],how="left", on="oncotech_id")
print("{} patients with clinical and radiation data for CHUM.".format(len(chum_merge)))
#chum_merge.drop(columns="oncotech_id", inplace=True)


223 patients with clinical and radiation data for CHUM.


In [98]:
chum_class = add_os_day_class(chum_merge, predictor)


/var/folders/km/l90xp47s19v3fplyb1b0bc480000gp/T/ipykernel_7178/2560263534.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["class_"+ predictor][df["class_"+ predictor] == 2] = 1
/var/folders/km/l90xp47s19v3fplyb1b0bc480000gp/T/ipykernel_7178/2560263534.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df["class_"+ predictor][df["class_"+ predictor] == 2] = 1
/var/folders/km/l90xp47s19v3fplyb1b0bc480000gp/T/ipykernel_7178/2560263534.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/

In [103]:
print(len(chum_class[["oncotech_id", "class_os_days"]]))
print(len(radomics_chum[["oncotech_id", "class_os_days"]]))

149
149


In [100]:
print(chum_class[["oncotech_id", "os_days", "class_os_days"]].sort_values("os_days"))
radomics_chum[["oncotech_id", "os_days", "class_os_days"]].sort_values("os_days")

    oncotech_id  os_days  class_os_days
133    236-CHUM        9              0
6      014-CHUM        9              0
57     114-CHUM       14              0
103    184-CHUM       16              0
79     150-CHUM       17              0
..          ...      ...            ...
86     157-CHUM     1760              1
23     044-CHUM     1818              1
5      012-CHUM     1888              1
107    188-CHUM     2047              1
55     111-CHUM     2047              1

[149 rows x 3 columns]


oncotech_id  os_days  class_os_days
20     236-CHUM        9              0
106    014-CHUM        9              0
101    114-CHUM       14              0
63     184-CHUM       16              0
78     150-CHUM       17              0
..          ...      ...            ...
16     157-CHUM     1760              1
46     044-CHUM     1818              1
90     012-CHUM     1888              1
72     111-CHUM     2047              1
100    188-CHUM     2047              1

[149 rows x 3 columns]

In [111]:
radomics_chum

index  Unnamed: 0_x oncotech_id center     sex         dob     age  \
0        2             4    004-CHUM   CHUM  Female  1950-06-17  67.964   
1       55            71    074-CHUM   CHUM  Female  1954-01-24  64.479   
2       43            53    056-CHUM   CHUM  Female  1967-09-12  51.053   
3      136           179    182-CHUM   CHUM    Male  1954-05-23  62.100   
4       85           115    118-CHUM   CHUM  Female  1944-01-10  74.897   
..     ...           ...         ...    ...     ...         ...     ...   
144     19            23    024-CHUM   CHUM  Female  1955-11-10  62.316   
145     65            88    091-CHUM   CHUM  Female  1930-10-02  87.003   
146     93           124    127-CHUM   CHUM  Female  1974-12-09  42.053   
147    104           138    141-CHUM   CHUM    Male  1955-03-17  62.048   
148    105           141    144-CHUM   CHUM    Male  1939-07-30  76.364   

     weight  height     bmi  ... wavelet.LLL_gldm_GrayLevelVariance  \
0      86.0     165  31.589  ...                         388.436840   
1      50.2     158  20.109  ...                         272.599934   
2      71.8     166  26.056  ...                         652.287673   
3      77.0     174  25.433  ...                         311.945511   
4      53.0     157  21.502  ...                         164.462886   
..      ...     ...     ...  ...                                ...   
144    49.3     154  20.788  ...                         281.817800   
145    37.0     155  15.401  ...                         331.037147   
146    66.3     170  22.941  ...                         513.292960   
147    64.0     178  20.199  ...                         391.585710   
148    77.0     180  23.765  ...                          12.642438   

    wavelet.LLL_gldm_HighGrayLevelEmphasis  \
0                              2035.413453   
1                              4358.013327   
2                              3728.897402   
3                              5925.705817   
4                             14590.842766   
..                                     ...   
144                            1371.450479   
145                            3892.745129   
146                            8683.414344   
147                            7369.685029   
148                             693.284384   

    wavelet.LLL_gldm_LargeDependenceEmphasis  \
0                                   2.804484   
1                                   3.609849   
2                                   1.968736   
3                                  11.932667   
4                                  10.400928   
..                                       ...   
144                                 2.386581   
145                                 6.580670   
146                                 7.174504   
147                                 7.793986   
148                                 5.571946   

     wavelet.LLL_gldm_LargeDependenceHighGrayLevelEmphasis  \
0                                          7829.888789       
1                                         17097.831145       
2                                          8411.616909       
3                                         70754.102596       
4                                        149183.834984       
..                                                 ...       
144                                        3620.000000       
145                                       33329.388932       
146                                       59604.283887       
147                                       57942.242275       
148                                        3902.372749       

    wavelet.LLL_gldm_LargeDependenceLowGrayLevelEmphasis  \
0                                             0.003951     
1                                             0.001162     
2                                             0.002049     
3                                             0.002268     
4                                             0.000740     
..              

In [102]:
compare = chum_class[["oncotech_id", "class_os_days", "os_days"]].merge(radomics_chum[["oncotech_id", "os_days", "class_os_days"]], on="oncotech_id", how="outer")
compare[comp]

oncotech_id  class_os_days_x  os_days_x  os_days_y  class_os_days_y
0      003-CHUM              1.0     1119.0     1119.0              1.0
1      004-CHUM              0.0      171.0      171.0              0.0
2      005-CHUM              1.0      985.0      985.0              1.0
3      007-CHUM              0.0      131.0      131.0              0.0
4      010-CHUM              1.0      922.0      922.0              1.0
..          ...              ...        ...        ...              ...
145    295-CHUM              0.0      104.0      104.0              0.0
146    296-CHUM              0.0      153.0      153.0              0.0
147    319-CHUM              1.0      740.0      740.0              1.0
148    322-CHUM              0.0      279.0      279.0              0.0
149    037-CHUM              NaN        NaN      604.0              1.0

[150 rows x 5 columns]

In [110]:
radomics_chum.oncotech_id.isin(["288-CHUM"]).any()

False

# Differences with Quartiles
Sevinj:
- use quantile function which returns the quantile as number
- distribute the original dataset into groups:
    - shortest = <= Q(0.33)
    - longest = >= Q(0.663)
- results in: excluding 288-CHUM with 183 days (shortest)
- including: 037-CHUM with 604 days to group longest

Elena:
- used qcut which directly assignes groups to the tertiles
- results in: excluding 037-CHUM with 183 days (longest)
- including: 288-CHUM with 604 days to group shortest


Same:
- same number of patients included in total:

Different:
- One patient with different group is in one group

In [106]:
compare[compare.class_os_days_x != compare.class_os_days_y]

oncotech_id  class_os_days_x  os_days_x  os_days_y  class_os_days_y
144    288-CHUM              0.0      283.0        NaN              NaN
149    037-CHUM              NaN        NaN      604.0              1.0

In [12]:
radomics_chum[["oncotech_id", "os_days"]]

oncotech_id  os_days
0      132-CHUM        0
1      067-CHUM        1
2      311-CHUM        1
3      014-CHUM        0
4      247-CHUM        1
..          ...      ...
143    016-CHUM        0
144    004-CHUM        0
145    198-CHUM        0
146    277-CHUM        0
147    194-CHUM        0

[148 rows x 2 columns]

In [ ]:
y = pd.DataFrame(radomics_chum[column_name])

X = radomics_chum[feature_name]

In [ ]:
df1.to_csv('first_chum_os.csv')
df3.to_csv('third_chum_os.csv')
df2.to_csv('middle_chum_os.csv')